modules/ROOT/pages/anypoint-mq/2.x/anypoint-mq-listener.adoc

```diff
@@ -230,6 +230,31 @@ In many cases there's a single common service between the processing of messages
 
 For this scenario, both subscribers stop polling for messages as soon as the error count reaches the `errorsThreshold="10"` value, counting both `FTP:RETRY_EXHAUSTED` and `HTTP:SERVICE_UNAVAILABLE` errors. When the `tripTimeout` elapses, one of the subscribers polls for a message and uses it to test the circuit, enabling the polling for both subscribers if the processing of that message succeeds.
 
+
+== FIFO Queues
+
+FIFO queues are most suitable for single-consumer scenarios. When one consumer is accessing a message, all other consumers are blocked until the first batch is processed. No messages are delivered until all in-flight messages are acknowledged or not acknowledged. With message groups, multiple consumers can access messages in a FIFO queue. In this case, a single consumer accesses messages in a group and message order is preserved within the message group.
```

Comment:
I had a moment's pause about "acknowledged or not acknowledged," wondering how new messages would know about not acknowledged.... Do you think it would be clearer to say "...until all in-flight messages are processed, either by being acknowledged or not acknowledged"?

Reply:
I also kind of wonder if we REALLY need to explain FIFO. If someone as nontechnical as I know what that means, then I am guessing your audience does.... I know there is other info related here, so ...maybe just a reduction?

Reply:
This is what we get when we can't use ACK/NACK.
We do need to explain FIFO _queues_ - how they behave - it's more than just first in first out.